### DISCUSSION TITLE
Being able to disable the folder based routing

### DISCUSSION THREAD

**Discussion by Luffyyy** [nuxt/nuxt#18248] [Ideas]
I hope that we'll be able to disable this, I prefer to do routing by myself and keep the folder structure cleaner. If I recall correctly, this wasn't available in nuxt 2.

Edit: it's sort of is possible, but it would make sense if there was an option to avoid it scanning it entirely.

**Comment by HenrijsS** [comment]
This was available in Nuxt 2.
Here's the module: https://github.com/nuxt-community/router-module
It allows you to rewrite the router to your needs.

I don't know about the compatabillity for Nuxt 3 yet.

**Reply by Luffyyy** [reply]
I meant more as an option and then just extend routes but thanks

**Reply by HenrijsS** [reply]
Maybe [this](https://nuxtjs.org/docs/configuration-glossary/configuration-router/#extendroutes) helps?

**Reply by Luffyyy** [reply]
Yeah I use this, just no option to just not automatically do the routes.

**Reply by HenrijsS** [reply]
I don't understand what you want then.
You can create your own routes with the module I specified or extend the already created routes from the docs.

**Reply by Luffyyy** [reply]
Correct, I can make them, but the idea is to be able to disable Nuxt's functionality of automatically adding what you have in /pages and letting the developer to do it themselves. Not being able to disable this means I have routes I didn't wish to add specifically.

**Reply by HenrijsS** [reply]
But that's exactly what the module does????

**Reply by Luffyyy** [reply]
Great, it won't hurt having it as an option without another module , which is what this idea is for.

**Reply by HenrijsS** [reply]
Oh. That's another story.
Automatic page routing is one of the reasons Nuxt exists. I believe that's why they don't focus on disabling it that much.

**Reply by Luffyyy** [reply]
Yeah fair, I use Nuxt for other reasons (notably SSR)

**Reply by HenrijsS** [reply]
That too.
But here's an alternative: https://ssr.vuejs.org/

**Comment by ivodolenc** [comment]
**Example of how to set routes manually in Nuxt2**

See the whole project [here](https://github.com/ivodolenc/nuxt-locomotive)

```js
// routes.js

const routes = [
  {
    name: 'home',
    path: '/',
    component: 'pages/home/index.vue'
  },
  {
    name: 'about',
    path: '/about',
    component: 'pages/about/index.vue'
  },
  {
    // ...
  }
]

export default routes
```

```js
// nuxt.config.js

import routes from './routes'

export default {
  router: {
    extendRoutes() {
      return routes
    }
  }
}
```
